### PR TITLE
new variant Seeed XIAO_ESP32S3

### PR DESCRIPTION
**Pull request by Maxwelltoo**
## Description of Change
new variant Seeed XIAO_ESP32S3

## Tests scenarios
I have tested my Pull Request on Arduino-esp32 core v2.0.2 with Seeed XIAO_ESP32S3 Board

### CODE REVIEW

File: boards.txt [hunk 1]
```diff
@@ -19779,7 +19779,7 @@ AirM2M_CORE_ESP32C3.menu.EraseFlash.all.upload.erase_cmd=-e
 
 XIAO_ESP32C3.name=XIAO_ESP32C3
 XIAO_ESP32C3.vid.0=0x2886
-XIAO_ESP32C3.pid.0=0x0047
+XIAO_ESP32C3.pid.0=0x0046
```
Comment (P-R-O-C-H-Y) [inline boards.txt:19782]: Just to be sure, is there any reason to change the PID of XIAO_ESP32C3?

File: boards.txt [hunk 2]
```diff
@@ -19931,6 +19931,188 @@ XIAO_ESP32C3.menu.EraseFlash.all.upload.erase_cmd=-e
 
 ##############################################################
 
+
+XIAO_ESP32S3.name=XIAO_ESP32S3
+XIAO_ESP32S3.vid.0=0x2886
+XIAO_ESP32S3.pid.0=0x0056
+XIAO_ESP32S3.vid.0=0x2886
+XIAO_ESP32S3.pid.0=0x8056
```
Comment (P-R-O-C-H-Y) [inline boards.txt:19939]: vid/pid is doubled. Delete one pair of it.

File: boards.txt [hunk 3]
```diff
@@ -19931,6 +19931,188 @@ XIAO_ESP32C3.menu.EraseFlash.all.upload.erase_cmd=-e
...
+
+XIAO_ESP32S3.menu.FlashSize.8M=8MB (64Mb)
+XIAO_ESP32S3.menu.FlashSize.8M.build.flash_size=8MB
+XIAO_ESP32S3.menu.FlashSize.8M.build.partitions=default_8MB
+
+XIAO_ESP32S3.menu.LoopCore.1=Core 1
+XIAO_ESP32S3.menu.LoopCore.1.build.loop_core=-DARDUINO_RUNNING_CORE=1
+XIAO_ESP32S3.menu.LoopCore.0=Core 0
+XIAO_ESP32S3.menu.LoopCore.0.build.loop_core=-DARDUINO_RUNNING_CORE=0
+XIAO_ESP32S3.menu.EventsCore.1=Core 1
```
Comment (P-R-O-C-H-Y) [inline boards.txt:20024]: Add empty line between LoopCore and EventsCore for better readability.

File: variants/XIAO_ESP32S3/pins_arduino.h [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+static const uint8_t SCK  = 7;
+
+static const uint8_t A0 = 1;
+static const uint8_t A1 = 2;
+static const uint8_t A2 = 3;
+static const uint8_t A3 = 4;
+static const uint8_t A4 = 5;
+static const uint8_t A5 = 6;
+// static const uint8_t A6 = 7; /* not pinout */
+// static const uint8_t A7 = 8;
```
Comment (P-R-O-C-H-Y) [inline variants/XIAO_ESP32S3/pins_arduino.h:40]: Remove unnecessary comments.

File: variants/XIAO_ESP32S3/pins_arduino.h [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+static const uint8_t T5 = 5;
+static const uint8_t T6 = 6;
+static const uint8_t T7 = 7;
+static const uint8_t T8 = 8;
+static const uint8_t T9 = 9;
+// static const uint8_t T10 = 10; /* not pinout */
+// static const uint8_t T11 = 11;
+// static const uint8_t T12 = 12;
+// static const uint8_t T13 = 13;
+// static const uint8_t T14 = 14;
```
Comment (P-R-O-C-H-Y) [inline variants/XIAO_ESP32S3/pins_arduino.h:70]: Remove unnecessary comments

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@Maxwelltoo please take a look on my comments.

File: boards.txt (hunk 1 above)
Comment (Maxwelltoo) [reply boards.txt:19782]: The previous PID was incorrect.

**Comment by Maxwelltoo** [comment]
@P-R-O-C-H-Y Thanks! I noticed these issues and addressed them one by one in the latest commit.

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Pull request merged**